testsuite/features/step_definitions/common_steps.rb

```diff
@@ -643,25 +643,31 @@
   $server.run("salt-key -y --accept=pxeboot.example.org")
 end
 
-When(/^I bootstrap (traditional|minion) client "([^"]*)" using bootstrap script with activation key "([^"]*)" from the proxy$/) do |client_type, host, key|
+When(/^I bootstrap (traditional|minion) client "([^"]*)" using bootstrap script with activation key "([^"]*)" from the server|proxy$/) do |client_type, host, key, target_type|
   # Preparation of bootstrap script for different types of clients
   client = client_type == 'traditional' ? '--traditional' : ''
+  # Use server if proxy is not defined as proxy is not mandatory
+  if target_type.include? 'server' || $proxy.nil?
+    target = $server
+  else
```

Comment:
I recommend here to use `or` instead of `||`:
`if target_type.include? 'server' or $proxy.nil?`

otherwise, you need to add parens (because `||` has different precedence than `or`):
`if (target_type.include? 'server') || $proxy.nil?`

Reply:
Thanks a lot. I will change it in a way you recommended.

Reply:
Done.